dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -157,8 +138,10 @@ class Recurrent[T : ClassTag] (
     var i = times
     while (i >= 1) {
       currentGradOutput(inputDim) = gradOutput.select(timeDim, i)
+      val _input = T(if (i > 1) modules(i - 2).output.toTable(hidDim)
+      else hidden, input.select(timeDim, i))
       currentGradOutput =
-        modules(i - 1).updateGradInput(inputs(i), currentGradOutput).toTable
+        modules(i - 1).updateGradInput(_input, currentGradOutput).toTable
```

Comment:
the i-th module input should be the (i-1)-th module output; when i decreases, the input should firstly contain previous module's output. As long as i decreases to 1, the hidden part of the input should be 0. The above code just did the converse operations...